packages/edit-site/src/components/block-list-exploded/item.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	store as blockEditorStore,
+	BlockPreview,
+	useBlockDisplayInformation,
+} from '@wordpress/block-editor';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { pure } from '@wordpress/compose';
+import { sprintf, __ } from '@wordpress/i18n';
+import { useMemo, useRef, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import BlockListExplodedTopToolbar from './top-toolbar';
+import { store as editSiteStore } from '../../store';
+
+function BlockListExplodedItem( { clientId } ) {
+	const blockWrapper = useRef();
+	const { block, isSelected } = useSelect(
+		( select ) => {
+			const { getBlock, isBlockSelected } = select( blockEditorStore );
+			return {
+				block: getBlock( clientId ),
+				isSelected: isBlockSelected( clientId ),
+			};
+		},
+		[ clientId ]
+	);
+	const { title } = useBlockDisplayInformation( clientId );
+	const { selectBlock } = useDispatch( blockEditorStore );
+	// If the exploded list becomes part of block-editor
+	// This mode also need to move into the block-editor store.
+	const { switchEditorMode } = useDispatch( editSiteStore );
+
+	// translators: %s: Type of block (i.e. Text, Image etc)
+	const blockLabel = sprintf( __( 'Block: %s' ), title );
+	const blocksToPreview = useMemo( () => [ block ], [ block ] );
+
+	useEffect( () => {
+		if ( isSelected ) {
+			blockWrapper.current.focus();
+		}
+	}, [ isSelected ] );
+
+	return (
+		<div
+			className={ classnames(
+				'edit-site-block-list-exploded__item-container',
+				{ 'is-selected': isSelected }
+			) }
+		>
+			{ isSelected && (
+				<BlockListExplodedTopToolbar clientId={ clientId } />
+			) }
+			<div
+				ref={ blockWrapper }
+				role="button"
+				onClick={ ( event ) => {
+					if ( event.detail === 1 ) {
+						selectBlock( clientId );
+					} else if ( event.detail === 2 ) {
+						switchEditorMode( 'visual' );
+					}
```

Comment:
I got a little confused here 😄  I assume `1` and `2` mean single and double click. Maybe we should use constants here.

Reply:
I can use booleans to make things clearer :)

Reply:
Thanks, Riad.